server/core/storage.go

```diff
@@ -362,27 +362,44 @@ func (s *Storage) Close() error {
 }
 
 // SaveGCSafePoint saves new GC safe point to storage.
-func (s *Storage) SaveGCSafePoint(safePoint uint64) error {
-	key := path.Join(gcPath, "safe_point")
+func (s *Storage) SaveGCSafePoint(serviceID string, safePoint uint64, ttl int64) error {
+	key := path.Join(gcPath, "safe_point", serviceID)
 	value := strconv.FormatUint(safePoint, 16)
-	return s.Save(key, value)
+	return s.SaveWithTTL(key, value, ttl)
 }
 
 // LoadGCSafePoint loads current GC safe point from storage.
+// It collects the all
 func (s *Storage) LoadGCSafePoint() (uint64, error) {
-	key := path.Join(gcPath, "safe_point")
-	value, err := s.Load(key)
+	safePoints, err := s.LoadAllGCSafePoint()
 	if err != nil {
-		return 0, err
-	}
-	if value == "" {
 		return 0, nil
 	}
-	safePoint, err := strconv.ParseUint(value, 16, 64)
+	min := uint64(0)
+	for _, val := range safePoints {
+		if min < safePoint {
+			min = safePoint
+		}
+	}
+	return min, nil
```

Comment:
when all safepoint exceeds TTL, it returns 0?

Reply:
I think so, TiDB would not set a TTL by default, so there is always a safe point.